fe1-web/model/objects/HDWallet.ts

```diff
@@ -98,30 +116,48 @@ export class HDWallet {
 
   /**
    * creates a new wallet object from the state (encryptedSeed)
-   * @param encryptedSeed wallet's encrypted seed
+   * @param encryptedSerializedSeed wallet's encrypted seed recovered from state
    * @return a new wallet object initialized from the given encrypted seed
    */
-  public async fromState(encryptedSeed: ArrayBuffer): Promise<HDWallet> {
-    await this.cryptoManager.initWalletStorage();
-    const walletFromState = new HDWallet();
-    walletFromState.setEncryptedSeed(encryptedSeed);
-    WalletStore.store(encryptedSeed);
-    return walletFromState;
+  public static async fromState(encryptedSerializedSeed: string): Promise<HDWallet> {
+    const wallet: HDWallet = new
+    HDWallet(HDWallet.deSerializeEncryptedSeed(encryptedSerializedSeed));
+
+    wallet.cryptoManager = new WalletCryptographyHandler();
+    await wallet.cryptoManager.initWalletStorage();
+    return wallet;
   }
 
   /**
    * returns the current wallet object to state (encryptedSeed)
+   * serialized as a String in order to be stored in redux state
    */
-  public toState(): ArrayBuffer {
-    return this.encryptedSeed;
+  public toState(): String {
+    return HDWallet.serializeEncryptedSeed(this.encryptedSeed);
```

Comment:
why not move the logic of that static function in here?
It's private, one line, and only used here.
You're saving yourself 1 indirection and improving the readability a lot.